NuMPI/IO/MPIFileIO.py

```diff
@@ -248,9 +250,10 @@ def _read_header(self):
             self.dtype = np.dtype(d['descr'])
             self.fortran_order = d['fortran_order']
             self.nb_grid_pts = d['shape']
+            self.data_start = self.file.Get_position()
         except Exception as err:
             # FIXME! This should be handled through a resource manager
-            if self.file is not None:
+            if self.file is not None and self.close_file_on_error:
                 self.file.Close()
```

Comment:
Is is correct to write "Close" with capital letter here? This looks a bit strange to me.
See https://docs.python.org/3.7/library/io.html?highlight=close#io.IOBase.close

Is there a test for this, I mean that closing works?

Reply:
it is the MPI Close. Or our stub implementation of it. https://mpi4py.readthedocs.io/en/stable/tutorial.html#mpi-io

Reply:
regarding test. We could test this in "test_corrupted_file" . But I don't know how to assert that the file has really been closed. Any idea ?

Reply:
Thanks, I haven't looked en that it is an MPI call. Hm .. the file protocol/interface in Python has the `closed` property for checking whether it is closed. If the MPI file doesn't have it - I don't know, maybe
```
with pytest.raises(???):
   <some code>
```

If this is too difficult now, let's continue here.